larq_compute_engine/core/packbits.h

```diff
@@ -382,19 +353,30 @@ inline void packbits_array(const TIn* input_array, const std::size_t n,
     memcpy(padding_buffer.data(), in, elements_left * sizeof(TIn));
     for (size_t i = elements_left; i < bitwidth; ++i)
       padding_buffer[i] = zero_point;
-    pack_bitfield<bitpack_order>(padding_buffer.data(), out, zero_point);
+    pack_bitfield(padding_buffer.data(), out, zero_point);
   }
 }
 
 // Bitpacks each row of a row-major matrix
-template <BitpackOrder bitpack_order, class TIn, class TOut_>
+template <class TIn, class TOut_>
 inline void packbits_matrix(const TIn* input, const std::size_t input_num_rows,
                             const std::size_t input_num_cols, TOut_* output,
                             const std::int32_t zero_point = 0) {
   // Force the types to be unsigned so that the function can be called on signed
   // types as well
   using TOut = typename std::make_unsigned<TOut_>::type;
 
+#ifdef __aarch64__
+  if (!CE_IS_BIG_ENDIAN && std::is_same<TIn, float>::value &&
```

Comment:
The optimised bitpacking is only correct on little-endian CPUs, hence this test here. That is I guess technically a regression as the previous optimised bitpacking I think did work on big-endian CPUs too (though we didn't test that behaviour).

I'm not sure it's worth bothering to support the faster code path on big-endian CPUs - a lot of effort for minimal gain. Thoughts?

Reply:
> I'm not sure it's worth bothering to support the faster code path on big-endian CPUs - a lot of effort for minimal gain. Thoughts?

I agree, but I wouldn't really know if we are excluding a major device here either. It looks like [Android is always little-endian](https://developer.android.com/ndk/guides/abis) and I think the Raspberry PIs are too, so I think we should be fine.

Reply:
I guess all that matters is that the weights and activations are bitpacked using the same code (regardless of what order that is). If we'd ever want to support big-endian CPUs, we would just add a `Prepare` step to unpack and repack the weights, then they'd always be in the native order of that platform, right?
I'm not sure if the old code would've worked for big-endian CPUs either.